tcav/imagenet_and_broden_fetcher.py

```diff
@@ -0,0 +1,268 @@
+  return urls_dataframe
+
+""" Downloads an image.
+
+Downloads and image from a image url provided and saves it under path.
+Filters away images that are corrupted or smaller than 10KB
+
+  Args:
+    path: Path to the folder where we're saving this image.
+    url: url to this image.
+
+  Raises:
+    Exception: Propagated from PIL.image.verify()
+"""
+
+
+def download_image(path, url):
+  image_name = url.split("/")[-1]
+  image_name = image_name.split("?")[0]
+  image_prefix = image_name.split(".")[0]
+  saving_path = os.path.join(path, image_prefix + ".jpg")
+  urllib.request.urlretrieve(url, saving_path)
+
+  try:
+    # Throw an exception if the image is unreadable or corrupted
+    Image.open(saving_path).verify()
+
+    # Remove images smaller than 10kb, to make sure we are not downloading empty/low quality images
+    if os.stat(saving_path).st_size < kMinFileSize:
+      tf.io.gfile.remove(saving_path)
+  # PIL.Image.verify() throws a default exception if it finds a corrupted image.
+  except Exception as e:
+    tf.io.gfile.remove(
+        saving_path
+    )  # We need to delete it, since urllib automatically saves them.
+    raise e
+
+
+""" For a imagenet label, fetches all URLs that contain this image, from the main URL contained in the dataframe
+
+  Args:
+    imagenet_dataframe: Pandas Dataframe containing the URLs for different
+      imagenet classes.
+    concept: A string representing Imagenet concept(i.e. "zebra").
+
+  Returns:
+    A list containing all urls for the imagenet label. For example
+    ["abc.com/image.jpg", "abc.com/image2.jpg", ...]
+
+  Raises:
+    tf.errors.NotFoundError: Error occurred when we can't find the imagenet
+    concept on the dataframe.
+"""
+
+
+def fetch_all_urls_for_concept(imagenet_dataframe, concept):
+  if imagenet_dataframe["class_name"].str.contains(concept).any():
+    all_images = imagenet_dataframe[imagenet_dataframe["class_name"] ==
+                                    concept]["url"].values[0]
+    bytes = urllib.request.urlopen(all_images)
+    all_urls = []
+    for line in bytes:
+      all_urls.append(line.decode("utf-8"))
+    return all_urls
+  else:
+    raise tf.errors.NotFoundError(
+        None, None, "Couldn't find any imagenet concept for " + concept +
+        ". Make sure you're getting a valid concept")
+
+
+####### Main methods
+""" For a given imagenet class, download images from the internet
+
+  Args:
+    path: String. Path where we're saving the data. Creates a new folder with
+      path/class_name.
+    class_name: String representing the name of the imagenet class.
+    number_of_images: Integer representing number of images we're getting for
+      this example.
+    imagenet_dataframe: Dataframe containing the URLs for different imagenet
+      classes.
+
+  Raises:
+    tf.errors.NotFoundError: Raised when imagenet_dataframe is not provided
+
+
+"""
+
+
+def fetch_imagenet_class(path="../tcav_working/zebra_scrape/",
```

Comment:
what is this default path exactly? not sure why there is a "../tcav_working" directory structure

Reply:
Removed the default settings